dev/PushNotifications/GetRawNotificationEventArgs.h

```diff
@@ -18,7 +18,10 @@ namespace winrt::Microsoft::Windows::PushNotifications
         {
             if (pair.Name() == L"payload")
             {
-                std::wstring payloadAsWstring{pair.Value()};
+                // Convert escaped components to its normal content
+                // from the conversion done in the LRP (see NotificationListener.cpp)
+                std::wstring payloadAsEscapedWstring{ pair.Value() };
+                std::wstring payloadAsWstring{ winrt::Windows::Foundation::Uri::UnescapeComponent(payloadAsEscapedWstring) };
```

Comment:
I would test the encoding decoding scenarios with payload limits < 8k. We should make sure that
a) Command Line is able to accept such large strings.
b) Any additional buffer from encoding Escape characters is accounted for in the command line
c) The Byte stream from the service is unaltered.

Reply:
Results:

**a)** WNS still rejects 8K payloads ☹️. 5K works okay.
**b) and c)** 5K of chinese characters were correctly handled, delivering the payload to the unpackaged app.